### PR TITLE
[SUCE-232] - Implement Redux and create frontend-builder container

### PR DESCRIPTION
**Pull request by TrebuhD**

### CODE REVIEW

File: client/elements/menus/language-base-menu.html [hunk 1]
```diff
@@ -7,32 +7,33 @@
 <link rel="import" href="../../bower_components/neon-animation/web-animations.html">
 <link rel="import" href="../../bower_components/app-storage/app-localstorage/app-localstorage-document.html">
```
Comment (gradam) [inline client/elements/menus/language-base-menu.html]: Those should be absolute paths. right  ?

File: client/redux-store.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+<!--
+ This element creates and describes the redux store.
+ ---
+ Redux introduction: http://redux.js.org/docs/introduction/
+ Polymer-redux docs: https://tur-nr.github.io/polymer-redux/
+ -->
+<script>
+    const initialState = {
+        currentLocation: 'homepage',
```
Comment (gradam) [inline client/redux-store.html]: What if the first page I visit is 'donation page' or anything other than homepage? For example from the link sent by someone to me.

File: client/redux-store.html [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        }
+    };
+
+    const store = Redux.createStore(
+        reducer,
+        initialState,
+        Redux.compose(
+            window.devToolsExtension
+                ? window.devToolsExtension()
+                : v => v
```
Comment (gradam) [inline client/redux-store.html]: semicolon

File: client/redux-store.html [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        )
+    );
+
+    ReduxMixin = PolymerRedux(store);
+</script>
```
Comment (gradam) [inline client/redux-store.html]: No newline at the end of the file

File: client/redux-store.html (hunk 3 above)
Comment (TrebuhD) [reply client/redux-store.html]: Nope, these are arguments

File: client/redux-store.html (hunk 2 above)
Comment (TrebuhD) [reply client/redux-store.html]: The basic process looks like this: 

the URL changes -> <sc-page-selector> detects this and updates the state -> the correct view is rendered.

For this case it's the same. The value is there to provide an idea of how the initial state tree would look, but it gets re-evaluated on every route change anyway.

File: client/elements/menus/language-base-menu.html (hunk 1 above)
Comment (TrebuhD) [reply client/elements/menus/language-base-menu.html]: correct!

**Pull request merged**